### PR TITLE
Skip obsolete payload save and cleanup for successful sync webhooks

### PR DESCRIPTION
**Pull request by cmiacz**
Backport of https://github.com/saleor/saleor/pull/16632 and all dependencies

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by korycins — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2686,3 +2688,9 @@ Just so you know, changes mentioned in this section are in a preview state and c
 - Add createdb note to getting started for Windows - #3106 by @ajostergaard
 - Update docs on pipenv - #3045 by @jxltom
 - Fixed a bug when saving webhook payload to Azure Storage - #16585 by @delemeator
+- Added validation for tax data received from tax app - #16720 by @zedzior
+- Fixed order-level discounts handling when using tax app for tax calculation - #16696 by @zedzior
+- Fixed bug when manual line discount doesn't override line-level vouchers - #16738 by @zedzior
+- Skipped obsolete payload save and cleanup for successfull sync webhooks - #16632 by @cmiacz
+- Skiped obsolete payload save and cleanup for successfull sync webhooks - #16632 by @cmmiacz
+- Skiped obsolete payload save and cleanup for successfull sync webhooks - #16632 by @cmiacz
```
Comment (NyanKiyoshi) [inline CHANGELOG.md:2696]: Fixes duplicate entries, typos, and PR number

```suggestion
- Skipped obsolete payload save and cleanup for successful sync webhooks - #16809 by @cmiacz
```

File: saleor/core/models.py [hunk 2]
```diff
@@ -189,6 +189,12 @@ def save_payload_file(self, payload_data: str):
         file_path = safe_join(prefix, file_name)
         self.payload_file.save(file_path, ContentFile(payload_bytes))
 
+    def save_as_file(self):
+        payload_data = self.payload
+        self.payload = ""
+        self.save()
```
Comment (NyanKiyoshi) [inline saleor/core/models.py:195]: Should it be like so instead? Not sure if the full save is intended or not?

```suggestion
        if self.payload != "":
            self.payload = ""
            self.save(update_fields=["payload"])
```

File: saleor/webhook/transport/utils.py [hunk 3]
```diff
@@ -409,42 +412,62 @@ def attempt_update(
...
 @allow_writer()
 def delivery_update(delivery: "EventDelivery", status: str):
     delivery.status = status
-    delivery.save(update_fields=["status"])
+    if delivery.id:
+        delivery.save(update_fields=["status"])
+
+
+@allow_writer()
+def save_unsuccessful_delivery_attempt(attempt: "EventDeliveryAttempt"):
```
Comment (NyanKiyoshi) [inline saleor/webhook/transport/utils.py:459]: I suggest renaming the function so it's clearer that it's a conditional save:

```suggestion
def maybe_save_unsuccessful_delivery_attempt(attempt: "EventDeliveryAttempt"):
```

**Review by NyanKiyoshi — APPROVED** [review]

File: saleor/core/models.py (hunk 2 above)
Comment (cmiacz) [reply saleor/core/models.py:195]: Full save is here on purpose as this methos should work also for unsaved objects
https://github.com/saleor/saleor/blob/b3f52b066aa682a8aad9508cfd33c4610e73fd84/saleor/webhook/transport/utils.py#L466

File: saleor/webhook/transport/utils.py (hunk 3 above)
Comment (cmiacz) [reply saleor/webhook/transport/utils.py:459]: But condition is based on status and it skips save only if it's sucessful so I guess the name is fine.
We can consider `maybe_save_delivery_attempt` or `save_delivery_attempt_if_not_sucessful`

**Pull request merged**